indexing-service/src/main/java/org/apache/druid/indexing/overlord/setup/DefaultWorkerBehaviorConfig.java

```diff
@@ -23,29 +23,52 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import org.apache.druid.indexing.overlord.autoscaling.AutoScaler;
 import org.apache.druid.indexing.overlord.autoscaling.NoopAutoScaler;
+import org.apache.druid.indexing.worker.config.WorkerConfig;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
 
 /**
+ * This configuration allows overlord to work with several autoscalers to run tasks of different categories.
  */
 public class DefaultWorkerBehaviorConfig implements WorkerBehaviorConfig
 {
-  private static final AutoScaler DEFAULT_AUTOSCALER = new NoopAutoScaler();
+  // Use the same category constant as for worker category to match default workers and autoscalers
+  public static final String DEFAULT_AUTOSCALER_CATEGORY = WorkerConfig.DEFAULT_CATEGORY;
+  private static final AutoScaler DEFAULT_AUTOSCALER = new NoopAutoScaler(DEFAULT_AUTOSCALER_CATEGORY);
 
   public static DefaultWorkerBehaviorConfig defaultConfig()
   {
-    return new DefaultWorkerBehaviorConfig(DEFAULT_STRATEGY, DEFAULT_AUTOSCALER);
+    return new DefaultWorkerBehaviorConfig(DEFAULT_STRATEGY, DEFAULT_AUTOSCALER, null);
   }
 
   private final WorkerSelectStrategy selectStrategy;
-  private final AutoScaler autoScaler;
+  private final List<AutoScaler> autoScalers;
 
   @JsonCreator
   public DefaultWorkerBehaviorConfig(
       @JsonProperty("selectStrategy") WorkerSelectStrategy selectStrategy,
-      @JsonProperty("autoScaler") AutoScaler autoScaler
+      @Deprecated @JsonProperty("autoScaler") AutoScaler autoScaler,
+      @JsonProperty("autoScalers") List<AutoScaler> autoScalers
   )
   {
     this.selectStrategy = selectStrategy;
-    this.autoScaler = autoScaler;
+    if (autoScaler != null && autoScalers != null) {
+      throw new IllegalArgumentException("The autoScaler and autoScalers properties are mutually exclusive");
+    }
+    if (autoScaler == null && autoScalers == null) {
+      throw new IllegalArgumentException("Either autoScaler or autoScalers property must be provided");
+    }
```

Comment:
I made this change but now I get a LGTM alert. It complains that the autoScalers variable can be null in line 64 which would lead to an NPE, but in my opinion LGTM is wrong. 
Can you advise?

Reply:
I agree, lgtm is just wrong in this case.

Also, I see that you  are using `==` instead of  `^`, while they both produce same result due to interning of boolean values I would prefer `^` which has  clearly defined XOR meaning and reader for the code  would  get immediately context\. `==` is "tricky" way to do what `^`  is doing.